qhana_plugin_registry/api/models/templates.py

```diff
@@ -54,9 +56,45 @@ class TemplateTabSchema(ApiObjectSchema):
     description = ma.fields.String(required=True, allow_none=False)
     location = ma.fields.String(required=True, allow_none=False, validate=Length(max=255))
     sort_key = ma.fields.Integer(required=True, allow_none=False, default=0)
-    filter_string = ma.fields.String(required=True, allow_none=False)
+    filter_string = ma.fields.String(required=True, allow_none=False, default="{}")
     plugins = ma.fields.Nested(ApiLinkSchema)
 
+    @staticmethod
+    def validate_filter(filter_dict: dict):
+        match filter_dict:
+            case {"and": l} | {"or": l}:
+                if not isinstance(l, list):
+                    raise ma.ValidationError("Invalid plugin filter.")
+                for f in l:
+                    TemplateTabSchema.validate_filter(f)
+            case {"not": f}:
+                TemplateTabSchema.validate_filter(f)
+            case {"name": f} | {"tag": f}:
+                if not isinstance(f, str):
+                    raise ma.ValidationError("Invalid plugin filter.")
```

Comment:
If you test for different errors, then try to give the users meaningful information what went wrong.
At least state that a "tag" or "name" condition was not a string (maybe include the whole condition in the error message or just the unexpected value).
Additionally, you could keep track of the path to the error and include that in the error message.
(This is relevant for all error messages in this method.)